### DISCUSSION TITLE
Can we use multiple ViewModel for same usercontrol WPF

### DISCUSSION THREAD

**Discussion by rajeshwaranpj** [reactiveui/ReactiveUI#3263] [Q&A] [answered]
IViewFor allows using a ViewModel to be rendered for that associated view.

Is it possible to render the same user control for different ViewModels?

Currently using this implementation below for example which registers one View to one ViewModel
```
Locator.CurrentMutable.RegisterLazySingleton(() => new HelpControl(), typeof(IViewFor<HelpViewModel>));

public partial class HelpControl: UserControl, IViewFor<HelpViewModel>
```

**Comment by ChrisPulman** [comment, marked as answer]
Hi, you can register multiple ViewModel's with the same common Interface, then register them individually decaring the context for each registration, you can then get the relevant ViewModel by passing the various context values to the Locator to retrieve your required ViewModel.
